microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceDiagnosticsParticipant.java

```diff
@@ -135,6 +138,17 @@ private void validateMethod(MethodDeclaration node, List<Diagnostic> diagnostics
 								diagnostics.add(d);
 							}
 						}
+					} else if (isMatchAnnotation(annotation, ASYNCHRONOUS_ANNOTATION)) {
+						Type methodReturnType = node.getReturnType2();
+						if (methodReturnType == null || methodReturnType instanceof Future) {
```

Comment:
I don't think this is what you want. java.lang.concurrent.Future is what users must ensure used as the method's return type. However, org.eclipse.jdt.core.dom.Type is not going to be an instance of that. It's going to be some object that maybe has a String value that represents "Future". I'd look for some fully qualified name of the type. That's what you want to compare against.

Reply:
Is there another way to cast/declare what the type is for the method?

Reply:
Yup. You could try :
- Finding all references to `getReturnType2()` in the codebase. This isn't the first time we've used it so we might have other usages where we need to compare the type name against something known. That might give some ideas on how it can be used.
- You could insert a breakpoint right where methodReturnType is assigned and inspect it at runtime. It might give some more ideas about the internal structure and whether it contains what you need.